cmake/cxx_std_check.cpp

```diff
@@ -0,0 +1,5 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+int main() { static_assert(__cplusplus >= CXX_STD); }
```

Comment:
[`static_assert`](https://en.cppreference.com/w/cpp/language/static_assert) should also have an error string (unless it's C++17+):
```suggestion
int main() { static_assert(__cplusplus >= CXX_STD, "Required C++ standard not supported."); }
```